src/scales.js

```diff
@@ -513,21 +517,22 @@ export function scale(options = {}) {
   return scale;
 }
 
-export function exposeScales(scaleDescriptors) {
+export function exposeScales(scales) {
   return (key) => {
     if (!registry.has((key = `${key}`))) throw new Error(`unknown scale: ${key}`);
-    return key in scaleDescriptors ? exposeScale(scaleDescriptors[key]) : undefined;
+    return scales[key];
   };
 }
 
 // Note: axis- and legend-related properties (such as label, ticks and
 // tickFormat) are not included here as they do not affect the scale’s behavior.
 function exposeScale({scale, type, domain, range, interpolate, interval, transform, percent, pivot}) {
-  if (type === "identity") return {type: "identity", apply: (d) => d, invert: (d) => d};
+  if (type === "identity")
+    return {type: "identity", apply: (d) => d, invert: (d) => d, ...(range !== undefined && {range})};
   const unknown = scale.unknown ? scale.unknown() : undefined;
   return {
     type,
-    domain: slice(domain), // defensive copy
+    domain: [...new Set(domain)], // defensive copy, ensure uniqueness
```

Comment:
I don’t think this is appropriate here; this only applies to ordinal scales, and we should do it earlier. I’ve noticed that we render duplicate elements in color legends when the domain repeats values, and that’s definitely a bug we should fix.